airbyte-integrations/connectors/source-klaviyo/source_klaviyo/source.py

```diff
@@ -21,7 +22,15 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, Any
             # we use metrics endpoint because it never returns an error
             _ = list(Metrics(api_key=config["api_key"]).read_records(sync_mode=SyncMode.full_refresh))
         except Exception as e:
-            return False, repr(e)
+            original_error_message = repr(e)
+
+            # Regular expression pattern to match the API key
+            pattern = r'api_key=\b\w+\b'
```

Comment:
I assume the API key can only be alphanumeric, because it's embedded in the URL. So this regex should be OK, although it doesn't allow for special characters.